File: FE/WEB/app/models/geo-country.js
import Model, { attr, belongsTo, hasMany } from '@ember-data/model';

export default class GeoCountryModel extends Model {
  @belongsTo('geo-region', { async: true, inverse: 'geoCountries' }) geoRegion;
  @belongsTo('geo-subregion', { async: true, inverse: 'GeoCountries' })
  geoSubregion;
  @hasMany('geo-city', { async: true, inverse: 'geoCountry' }) geoCities;
  @hasMany('geo-first-division', { async: true, inverse: 'geoCountry' })
  geoFirstDivisions;
  @hasMany('geo-second-division', { async: true, inverse: 'geoCountry' })
  geoSecondDivisions;
  @hasMany('geo-third-division', { async: true, inverse: 'geoCountry' })
  geoThirdDivisions;
  @hasMany('geo-mapping', { async: true, inverse: 'geoCountry' }) geoMappings;

  @attr('string', { defaultValue: '' }) name;
  @attr('string', { defaultValue: '' }) iso3;
  @attr('string', { defaultValue: '' }) numericCode;
  @attr('string', { defaultValue: '' }) iso2;
  @attr('string', { defaultValue: '' }) phoneCode;
  @attr('string', { defaultValue: '' }) capital;
  @attr('string', { defaultValue: '' }) currency;
  @attr('string', { defaultValue: '' }) currencyName;
  @attr('string', { defaultValue: '' }) currencySymbol;
  @attr('string', { defaultValue: '' }) internetDomain;
  @attr('string', { defaultValue: '' }) native;
  @attr('number', { defaultValue: 0 }) geoRegionId;
  @attr('number', { defaultValue: 0 }) geoSubregionId;
  @attr('string', { defaultValue: '' }) timeZones;
  @attr('string', { defaultValue: '' }) translations;
  @attr('number', { defaultValue: 0 }) numberOfDivisions;
}

File: BE/Src/MIT.Fwk.Infrastructure/Entities/GeoCountry.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using System.Collections.Generic;

namespace MIT.Fwk.Infrastructure.Entities
{
    [Resource]
    public class GeoCountry : Identifiable<int>
    {
        [Attr]
        public string Name { get; set; }
        [Attr]
        public string Iso3 { get; set; }
        [Attr]
        public string NumericCode { get; set; }
        [Attr]
        public string Iso2 { get; set; }
        [Attr]
        public string PhoneCode { get; set; }
        [Attr]
        public string Capital { get; set; }
        [Attr]
        public string Currency { get; set; }
        [Attr]
        public string CurrencyName { get; set; }
        [Attr]
        public string CurrencySymbol { get; set; }
        [Attr]
        public string InternetDomain { get; set; }
        [Attr]
        public string Native { get; set; }
        [Attr]
        public int? GeoRegionId { get; set; }
        [Attr]
        public int? GeoSubregionId { get; set; }
        [Attr]
        public string TimeZones { get; set; }
        [Attr]
        public string Translations { get; set; }
        [Attr]
        public int? NumberOfDivisions { get; set; }

        [HasOne]
        public virtual GeoRegion GeoRegion { get; set; }

        [HasOne]
        public virtual GeoSubregion GeoSubregion { get; set; }
        [HasMany]
        public virtual ICollection<GeoCity> GeoCities { get; set; }
        [HasMany]
        public virtual ICollection<GeoFirstDivision> GeoFirstDivisions { get; set; }
        [HasMany]
        public virtual ICollection<GeoSecondDivision> GeoSecondDivisions { get; set; }
        [HasMany]
        public virtual ICollection<GeoThirdDivision> GeoThirdDivisions { get; set; }
        [HasMany]
        public virtual ICollection<GeoMapping> GeoMappings { get; set; }
    }
}
